spec/lumberjack_spec.rb

```diff
@@ -76,10 +76,12 @@
     # TODO(sissel): Make sure this doesn't take forever, do a timeout.
     count = 0
     events = []
-    connection = server.accept
-    connection.run do |event|
+    server.run do |event|
       events << event
-      connection.close if events.length == lines.length
+      if events.length == lines.length
+        server.stop
+        raise EOFError.new("Force the connection out of the read loop")
```

Comment:
I moved away from `server.run` because the semantics of controlling the server or aborting the connection were too intertwined. Is there a reason we need `server.run` here? Or can you just do `server.accept` to get a connection and then close it and the server separately?